From 61e6da4fdfc7734cf66783296dded77f5c2d0c4c Mon Sep 17 00:00:00 2001
From: Vishal Sagar <vishal.sagar@xilinx.com>
Date: Thu, 14 Mar 2019 03:55:53 -0700
Subject: [PATCH 0567/1851] dt-bindings: media: vpss-csc: Make max width and
 height mandatory properties

commit d471d10cc36b66d8fcb8e45cf17411d656c9b984 from
https://github.com/Xilinx/linux-xlnx.git

This patch makes the xlnx,max-width and xlnx,max-height mandatory dt
properties. This breaks backward compatibility.

Signed-off-by: Vishal Sagar <vishal.sagar@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 .../bindings/media/xilinx/xlnx,v-vpss-csc.txt       | 13 ++++++++-----
 1 file changed, 8 insertions(+), 5 deletions(-)

diff --git a/Documentation/devicetree/bindings/media/xilinx/xlnx,v-vpss-csc.txt b/Documentation/devicetree/bindings/media/xilinx/xlnx,v-vpss-csc.txt
index 491f7dbd2b61..b3627af85e6e 100644
--- a/Documentation/devicetree/bindings/media/xilinx/xlnx,v-vpss-csc.txt
+++ b/Documentation/devicetree/bindings/media/xilinx/xlnx,v-vpss-csc.txt
@@ -11,6 +11,12 @@ Required properties:
 
 - clocks: Reference to the clock that drives the ap_clk signal.
 
+- xlnx,max-height: Maximum number of lines.
+  Valid range from 64 to 4320.
+
+- xlnx,max-width: Maximum number of pixels in a line.
+  Valid range from 64 to 8192.
+
 - reset-gpios: Specifier for a GPIO that assert VPSS CSC (AP_RST_N) reset.
 
 - ports: Video ports, using the DT bindings defined in ../video-interfaces.txt.
@@ -24,17 +30,14 @@ Required port properties:
 
 - xlnx,video-width: Video width as defined in video.txt. Must be either 8 or 10.
 
-Optional properties:
-
-- xlnx,max-height: Maximum number of lines.
-- xlnx,max-width: Maximum number of pixels in a line.
-
 Example:
 	csc_1:csc@a0040000 {
 		compatible = "xlnx,v-vpss-csc";
 		reg = <0x0 0xa0040000 0x0 0x10000>;
 		clocks = <&vid_stream_clk>;
 		reset-gpios = <&gpio 84 1>;
+		xlnx,max-width = <3840>;
+		xlnx,max-height = <2160>;
 
 		ports {
 			#address-cells = <1>;
-- 
2.31.1

